pinot-core/src/main/java/org/apache/pinot/core/operator/combine/GroupByOrderByCombineOperator.java

```diff
@@ -242,7 +255,7 @@ public void runJob() {
       // Set the execution statistics.
       CombineOperatorUtils.setExecutionStatistics(mergedBlock, _operators);
 
-      if (_indexedTable.size() >= _indexedTableCapacity) {
+      if (_indexedTable.size() >= _trimSize) {
```

Comment:
(Critical) This is incorrect. Indexed table size can be the same as trim size in regular cases.

Reply:
this has always been incorrect (inaccurate actually) for both SQL and PQL. Will fix in a follow-up. Added a TODO

Reply:
It is correctly set for PQL. Please remove this if block, and maybe add a TODO to properly set this flag